tests/ImageSharp.Tests/Processing/Processors/Transforms/ResizeTests.cs

```diff
@@ -216,6 +216,32 @@ public void Resize_DoesNotBleedAlphaPixels<TPixel>(TestImageProvider<TPixel> pro
                 appendSourceFileOrDescription: false);
         }
 
+        [Theory]
+        [WithFile(TestImages.Png.Kaboom, DefaultPixelType, false)]
+        [WithFile(TestImages.Png.Kaboom, DefaultPixelType, true)]
+        public void Resize_PremultiplyAlpha<TPixel>(TestImageProvider<TPixel> provider, bool premultiplyAlpha)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            string details = premultiplyAlpha ? "On" : "Off";
+
+            provider.RunValidatingProcessorTest(
+                x =>
+                {
+                    var resizeOptions = new ResizeOptions()
+                    {
+                        Size = x.GetCurrentSize() / 2,
+                        Mode = ResizeMode.Crop,
+                        Sampler = KnownResamplers.Bicubic,
+                        Compand = false,
+                        PremultiplyAlpha = premultiplyAlpha
+                    };
+                    x.Resize(resizeOptions);
```

Comment:
Should there be a ResizeExtension added for quickly specifying PremultiplyAlpha?

Also, here are the reference outputs:
[ResizeTests_PremultiplyAlpha_RefOutput.zip](https://github.com/SixLabors/ImageSharp/files/5806355/ResizeTests_PremultiplyAlpha_RefOutput.zip)

Reply:
Since this will be a rarely used option, I prefer not to add new extension methods. There are too many permutations, `ResizeOptions` is here so we can keep them under control.